### PR TITLE
Fix dedendum calculation

### LINKED ISSUE #38: Dedendum calculation is wrong
image

```
m=3
n1=20
n2=8
d=m*(n1+n2)/2
g1 = SpurGear(
        module=m,
        tooth_count=n1,
        pressure_angle=14.5,
        root_fillet=0.5 * MM,
        thickness=5 * MM,
)
g2 = Location((d,0,0))*Rot(0,0,360/n2/2)*SpurGear(
        module=m,
        tooth_count=n2,
        pressure_angle=14.5,
        root_fillet=0.5 * MM,
        thickness=5 * MM,
)
```

### PR DESCRIPTION
**Pull request by smurfix**
Closes #38.

### CODE REVIEW

**Comment by smurfix** [comment]
Tested with various gear ratios and modules.

File: src/bd_warehouse/gear.py [hunk 1]
```diff
@@ -105,7 +105,8 @@ def __init__(
             r = self.base_radius + involute_size * i / 10
             α = acos(self.base_radius / r)  # in radians
             involute = tan(α) - α
-            pnts.append((r * cos(involute), r * sin(involute)))
+            if (rp := r * cos(involute)) > self.root_radius:
```
Comment (gumyr) [inline src/bd_warehouse/gear.py:108]: I believe this check will always pass as `r` is at least the `base_radius`.

File: src/bd_warehouse/gear.py [hunk 2]
```diff
@@ -120,7 +121,8 @@ def __init__(
                 Vector(self.addendum_radius, 0),
                 -self.addendum_radius,
             )
-            fillet(tooth.vertices().sort_by(Axis.X)[1], root_fillet)
+            if root_fillet:
```
Comment (gumyr) [inline src/bd_warehouse/gear.py:124]: Nice.  I'm not sure one would want to actually create a gear without a root fillet but there doesn't seem to be a reason to force it.

File: src/bd_warehouse/gear.py [hunk 3]
```diff
@@ -93,7 +93,7 @@ def __init__(
         self.dedendum = (
             dedendum
             if dedendum is not None
-            else (self.pitch_radius - self.base_radius) + 2 * root_fillet
+            else (1.2*module + root_fillet)
```
Comment (gumyr) [inline src/bd_warehouse/gear.py:96]: After reviewing this it seems like the value should just be `1.25 * module` - where `1.25` is supposedly a common value.  The `root_fillet` won't interfere unless very large values are used.

**Review by gumyr — CHANGES_REQUESTED** [review]
Thank you for pointing this out - a pretty significant problem.

File: src/bd_warehouse/gear.py (hunk 1 above)
Comment (smurfix) [reply src/bd_warehouse/gear.py:108]: It doesn't. See the second image in my bug report. Or simply create a gear with 100 teeth, which will crash trying to do the root fillet, which is how I found that problem.

File: src/bd_warehouse/gear.py (hunk 3 above)
Comment (smurfix) [reply src/bd_warehouse/gear.py:96]: That should work too. Will change.

**Comment by gumyr** [comment]
Thanks!  The 100 tooth gear is really an invalid gear specification - normally a pressure angle of 20 degrees are used for these high tooth count gears.

**Comment by smurfix** [comment]
Might be a good idea to emit a warning then. "Insufficient gear depth - increase pressure angle"?

**Pull request merged**